BootloaderCommonPkg/Library/ShellLib/CmdLog.c

```diff
@@ -0,0 +1,104 @@
+/** @file
+  Shell command `log` to print the contents of the log buffer.
+
+  Copyright (c) 2018, Intel Corporation. All rights reserved.<BR>
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php.
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include <Library/ShellLib.h>
+#include <Library/DebugLib.h>
+#include <Library/HobLib.h>
+#include <Guid/LoaderFspInfoGuid.h>
+#include <Library/ConsoleInLib.h>
+#include <Library/ConsoleOutLib.h>
+#include <Library/BootloaderCommonLib.h>
+
+/**
+  Print the contents of the log buffer
+
+  @param[in]  Shell        shell instance
+  @param[in]  Argc         number of command line arguments
+  @param[in]  Argv         command line arguments
+
+  @retval EFI_SUCCESS
+
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+ShellCommandLogFunc (
+  IN SHELL  *Shell,
+  IN UINTN   Argc,
+  IN CHAR16 *Argv[]
+  );
+
+CONST SHELL_COMMAND ShellCommandLog = {
+  L"log",
+  L"Print messages stored in boot log buffer",
+  &ShellCommandLogFunc
+};
+
+/**
+  Print the contents of the log buffer
+
+  @param[in]  Shell        shell instance
+  @param[in]  Argc         number of command line arguments
+  @param[in]  Argv         command line arguments
+
+  @retval EFI_SUCCESS
+
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+ShellCommandLogFunc (
+  IN SHELL  *Shell,
+  IN UINTN   Argc,
+  IN CHAR16 *Argv[]
+  )
+{
+  DEBUG_LOG_BUFFER_HEADER *LogBufHdr;
+  CONST UINTN              LinesPerPage = 30;
```

Comment:
Should we define a PCD for this? 
Ideally it should match the console row number dynamically.

Reply:
I prefer we keep it like this for now and add / replace this with a generic paging routine in console lib in a follow-up PR which can handle single line feeding and page down functions. I find it useful to use fewer than total screen rows so the reader has prior context around the new messages.

Reply:
OK, I am OK with it.  
To conform to coding style, can you move this const into global const variable instead ?

Reply:
Fixed